doc/neps/nep-0035-array-creation-dispatch-with-array-function.rst

```diff
@@ -8,16 +8,176 @@ NEP 35 — Array Creation Dispatching With __array_function__
 :Status: Draft
 :Type: Standards Track
 :Created: 2019-10-15
-:Updated: 2020-08-06
+:Updated: 2020-08-17
 :Resolution:
 
 Abstract
 --------
 
 We propose the introduction of a new keyword argument ``like=`` to all array
-creation functions to permit dispatching of such functions by the
-``__array_function__`` protocol, addressing one of the protocol shortcomings,
-as described by NEP-18 [1]_.
+creation functions, this argument permits the creation of an array based on
+a non-NumPy reference array passed via that argument, resulting in an array
+defined by the downstream library implementing that type, which also implements
+the ``__array_function__`` protocol. With this we address one of that
+protocol's shortcomings, as described by NEP 18 [1]_.
```

Comment:
```suggestion
creation functions to address one of the shortcomings of `__array_function__`,
(see NEP 18 [1]_). The `like` kwarg will create an instance of the type of the
argument, enabling direct creation of non-NumPy arrays. The target array type
must implement the ``__array_function__`` protocol.
```

Reply:
I did a few changes to your suggestion, but adopted the new structure you suggested in https://github.com/numpy/numpy/pull/17093/commits/69e3e7186a39abeead5adf8edba5a24d71e96cb9 . Thanks @mattip.